functions-and-scope.js

```diff
@@ -53,31 +87,78 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 // averageGrade([6, 4, 5]) geeft xxxx
 // averageGrade([8, 9, 4, 6, 10]) geeft xxxx
 
+function averageGrade(gradesArray) {
+    let sumOfGrades = 0
+    for (let i = 0; i < gradesArray.length; i++) {
+        sumOfGrades = sumOfGrades + gradesArray[i];
+        }
+    return sumOfGrades / gradesArray.length;
+}
+
+const averageGrade1 = averageGrade(grades);
+const averageGrade2 = averageGrade([6, 4, 5]);
+const averageGrade3 = averageGrade([8, 9, 4, 6, 10]);
+
+console.log(averageGrade1, averageGrade2, averageGrade3)
+
 
 /* 2c: Afronden op twee decimalen */
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
+const averageGrade1Rounded = averageGrade1.toFixed(2);
+const averageGrade2Rounded = averageGrade2.toFixed(2);
+const averageGrade3Rounded = averageGrade3.toFixed(2);
+
+console.log(averageGrade1Rounded, averageGrade2Rounded, averageGrade3Rounded);
 
+// Via Google kwam ik op de toFixed() methode. Het lukte me alleen op deze manier om het werkend te krijgen. Ik heb geprobeerd of ik dit ook ergens in de functie kon verwerken,
+// of in de declaratie van de const variabelen, maar dat lukte me niet. Is dit wel mogelijk?
 
 
 /* Bonusopdracht: hoogste cijfer */
 
 /* 3a: Script schrijven  */
 // Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
-// * Hoe kan ik iedere waarde van de array langsgaan?
-// * Op welke conditie moet ik checken?
-// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+// * Hoe kan ik iedere waarde van de array langsgaan? Met een for-loop. zoals bij de opdrachten hierboven al eerder is gedaan.
+// * Op welke conditie moet ik checken? Of het huidige cijfer hoger is dan de hoogst opgeslagen waarde/
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan? Door vooraf een let variabele te declareren om het cijfer in op te slaan.
 // Log het antwoord in de terminal.
 
 // ---- Verwachte uitkomst: 9
 
+let highestGrade = 0;
```

Comment:
Het is geen goed voorbeeld, maar ik wil het je wel meegeven:
Wanneer je een array meekrijgt ([-1, -3, -2]) met alleen maar negatieve getallen, dan zal de functie altijd 0 teruggeven. Het is daarom verstandig om altijd de eerste waarde in de array te pakken:
```suggestion
let highestGrade = grades[0];
```